src/Java.Interop.Tools.JavaSource/Java.Interop.Tools.JavaSource/SourceJavadocToXmldocGrammar.HtmlBnfTerms.cs

```diff
@@ -102,41 +101,37 @@ internal void CreateRules (SourceJavadocToXmldocGrammar grammar)
 
 				InlineHyperLinkDeclaration.Rule = InlineHyperLinkOpenTerm + InlineDeclarations + CreateEndElement ("a", grammar, optional: true);
 				InlineHyperLinkDeclaration.AstConfig.NodeCreator = (context, parseNode) => {
-					var unparsedAElementValue = string.Empty;
-					foreach (var cn in parseNode.ChildNodes) {
-						if (cn.ChildNodes?.Count > 1) {
-							foreach (var gcn in cn.ChildNodes) {
-								unparsedAElementValue += gcn.AstNode?.ToString ();
-							}
-						} else {
-							unparsedAElementValue += cn.AstNode?.ToString ();
-						}
-					}
+					var nodesAsString = GetChildNodesAsString (parseNode);
+					var tokenValue = parseNode.ChildNodes [0].Token.Text;
+					int stopIndex = nodesAsString.IndexOf ('>');
 
-					var seeElement = TryParseHRef (unparsedAElementValue);
-					if (seeElement == null)
-						seeElement = TryParseHRef (WebUtility.HtmlDecode (unparsedAElementValue), logError: true);
+					if (stopIndex == -1 || !tokenValue.Contains ("href", StringComparison.OrdinalIgnoreCase)) {
+						parseNode.AstNode = new XText (nodesAsString);
+						return;
+					}
 
-					var hrefValue = seeElement?.Attribute ("href")?.Value ?? string.Empty;
-					if (!string.IsNullOrEmpty (hrefValue) &&
-							(hrefValue.StartsWith ("http", StringComparison.OrdinalIgnoreCase) || hrefValue.StartsWith ("www", StringComparison.OrdinalIgnoreCase))) {
-						parseNode.AstNode = seeElement;
+					var attributeName = parseNode.ChildNodes [0].Term.Name;
+					var attributeValue = nodesAsString.Substring (0, stopIndex).Trim ().Trim ('\'', '"');
+					var elementValue = nodesAsString.Substring (stopIndex + 1);
+					if (!string.IsNullOrEmpty (attributeValue) &&
+						(attributeValue.StartsWith ("http", StringComparison.OrdinalIgnoreCase) || attributeValue.StartsWith ("www", StringComparison.OrdinalIgnoreCase))) {
```

Comment:
Code convention: when an `if` condition spans multiple lines, the condition should be indented an additional tab stop, so that it doesn't appear to be related to the contents of the `if` block:

```csharp
if (!string.IsNullOrEmpty (attributeValue) &&
        (attributeValue.StartsWith ("http", StringComparison.OrdinalIgnoreCase) || attributeValue.StartsWith ("www", StringComparison.OrdinalIgnoreCase))) {
    var unparsed = $"<see href=\"{attributeValue}\">{elementValue}</see>";
    // …
}
```

Reply:
I'm not certain about hardcoding starting with `www` if `http` isn't used.

If I look at the API-33 Java sources:

```
% grep -ri '<a href.*' --include=\*.java | grep -v http
```

I see such "interesting" matches as:

  * `{@docRoot}`!

        ./androidx/renderscript/Element.java: * <a href="{@docRoot}guide/topics/renderscript/index.html">RenderScript</a>

  * FTP!

        ./org/w3c/dom/ls/LSOutput.java:     * Assigned Numbers Authority [<a href='ftp://ftp.isi.edu/in-notes/iana/assignments/character-sets'>IANA-CHARSETS</a>]

  * Member cross references?!

        ./org/xmlpull/v1/XmlPullParser.java: * <dt><a href="#START_TAG">START_TAG</a><dd> An XML start tag was read.

  * Relative paths

        ./java/net/URLEncoder.java:     *    <a href="../lang/package-summary.html#charenc">character

  * Links spanning multiple lines, e.g. from `java/net/URLEncoder.java`:

    ```
     * <em><strong>Note:</strong> The <a href=
     * "http://www.w3.org/TR/html40/appendix/notes.html#non-ascii-chars">
     * World Wide Web Consortium Recommendation</a>
    ```

    (Maybe this is already supported; I just found it "interesting" to see `<a href=` and nothing after it in the `grep` output…)

Maybe ignoring all of these is appropriate.

But what got me started looking at this was the "attribute values starting with `www` that *don't* use `http`" bit, and I can't find anything matching that in the Android API-33 docs.  I'm not sure that we should special-case for `www`.

Reply:
I've removed the `www` check. The docRoot and multiline cases mentioned above are converted as expected, and cross references will need to be converted as part of future cref work.